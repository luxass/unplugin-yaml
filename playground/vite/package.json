{
  "name": "playground",
  "version": "0.0.0",
  "type": "module",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview",
    "build-inspect": "serve .vite-inspect"
  },
  "dependencies": {
    "solid-js": "^1.8.18"
  },
  "devDependencies": {
    "typescript": "^5.5.2",
    "unplugin-yaml": "workspace:*",
    "vite": "^5.3.2",
    "vite-plugin-inspect": "^0.8.4",
    "vite-plugin-solid": "^2.10.2"
  }
}
