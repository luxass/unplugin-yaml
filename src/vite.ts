/**
 * This entry file is for Vite plugin.
 *
 * @module
 */

import { createVitePlugin } from "unplugin";
import { unpluginFactory } from "./";

/**
 * Vite plugin
 *
 * @example
 * ```ts
 * // vite.config.ts
 * import yaml from "unplugin-yaml/vite"
 *
 * export default defineConfig({
 *   plugins: [yaml()],
 * })
 * ```
 */
export default createVitePlugin(unpluginFactory);
