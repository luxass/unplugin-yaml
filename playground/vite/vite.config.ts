import { defineConfig } from "vite";
import solid from "vite-plugin-solid";
import Inspect from "vite-plugin-inspect";
import YamlPlugin from "unplugin-yaml/vite";

export default defineConfig({
  plugins: [
    solid(),
    Inspect({
      build: true,
    }),
    YamlPlugin(),
  ],
});
