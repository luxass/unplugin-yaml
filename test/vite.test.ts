import { join } from "node:path";
import { build } from "vite";
import { describe, expect, it } from "vitest";
import { testdir } from "vitest-testdirs";
import YAMLPlugin from "../src/vite";

describe("handles yaml", () => {
  it("expect yaml import to be a json object", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yaml"));

    expect(testdirPath).toBeDefined();

    const result = await build({
      build: {
        lib: {
          entry: join(testdirPath, "basic.js"),
          formats: ["es"],
          fileName: "bundle",
          name: "bundle",
        },
        outDir: join(testdirPath, "dist"),
        minify: false,
      },
      plugins: [YAMLPlugin()],
    });

    if (!Array.isArray(result)) {
      expect.fail("result is not an array");
    }

    expect(result).toBeDefined();

    const firstResult = result[0];

    expect(firstResult).toBeDefined();
    expect(firstResult?.output).toBeDefined();
    expect(firstResult?.output[0]).toBeDefined();
    expect(firstResult?.output[0].code).toBeDefined();
    expect(firstResult?.output[0].code).toMatchSnapshot();
  });

  it("expect yaml import to be a string", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yaml"));

    expect(testdirPath).toBeDefined();

    const result = await build({
      build: {
        lib: {
          entry: join(testdirPath, "basic-raw.js"),
          formats: ["es"],
          fileName: "bundle",
          name: "bundle",
        },
        outDir: join(testdirPath, "dist"),
        minify: false,
      },
      plugins: [YAMLPlugin()],
    });

    if (!Array.isArray(result)) {
      expect.fail("result is not an array");
    }

    expect(result).toBeDefined();

    const firstResult = result[0];

    expect(firstResult).toBeDefined();
    expect(firstResult?.output).toBeDefined();
    expect(firstResult?.output[0]).toBeDefined();
    expect(firstResult?.output[0].code).toBeDefined();
    expect(firstResult?.output[0].code).toMatchSnapshot();
  });
});

describe("handles yml", () => {
  it("expect yml import to be a json object", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yml"));

    expect(testdirPath).toBeDefined();

    const result = await build({
      build: {
        lib: {
          entry: join(testdirPath, "basic.js"),
          formats: ["es"],
          fileName: "bundle",
          name: "bundle",
        },
        outDir: join(testdirPath, "dist"),
        minify: false,
      },
      plugins: [YAMLPlugin()],
    });

    if (!Array.isArray(result)) {
      expect.fail("result is not an array");
    }

    expect(result).toBeDefined();

    const firstResult = result[0];

    expect(firstResult).toBeDefined();
    expect(firstResult?.output).toBeDefined();
    expect(firstResult?.output[0]).toBeDefined();
    expect(firstResult?.output[0].code).toBeDefined();
    expect(firstResult?.output[0].code).toMatchSnapshot();
  });

  it("expect yml import to be a string", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yml"));

    expect(testdirPath).toBeDefined();

    const result = await build({
      build: {
        lib: {
          entry: join(testdirPath, "basic-raw.js"),
          formats: ["es"],
          fileName: "bundle",
          name: "bundle",
        },
        outDir: join(testdirPath, "dist"),
        minify: false,
      },
      plugins: [YAMLPlugin()],
    });

    if (!Array.isArray(result)) {
      expect.fail("result is not an array");
    }

    expect(result).toBeDefined();

    const firstResult = result[0];

    expect(firstResult).toBeDefined();
    expect(firstResult?.output).toBeDefined();
    expect(firstResult?.output[0]).toBeDefined();
    expect(firstResult?.output[0].code).toBeDefined();
    expect(firstResult?.output[0].code).toMatchSnapshot();
  });
});

it("handle multi document", async () => {
  const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/multi"));

  expect(testdirPath).toBeDefined();

  const result = await build({
    build: {
      lib: {
        entry: join(testdirPath, "multi.js"),
        formats: ["es"],
        fileName: "bundle",
        name: "bundle",
      },
      outDir: join(testdirPath, "dist"),
      minify: false,
    },
    plugins: [YAMLPlugin({
      type: "multi",
    })],
  });

  if (!Array.isArray(result)) {
    expect.fail("result is not an array");
  }

  expect(result).toBeDefined();

  const firstResult = result[0];

  expect(firstResult).toBeDefined();
  expect(firstResult?.output).toBeDefined();
  expect(firstResult?.output[0]).toBeDefined();
  expect(firstResult?.output[0].code).toBeDefined();
  expect(firstResult?.output[0].code).toMatchSnapshot();
});

it("handle transforms", async () => {
  const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/transform"));

  expect(testdirPath).toBeDefined();

  const result = await build({
    build: {
      lib: {
        entry: join(testdirPath, "transform.js"),
        formats: ["es"],
        fileName: "bundle",
        name: "bundle",
      },
      outDir: join(testdirPath, "dist"),
      minify: false,
    },
    plugins: [YAMLPlugin({
      transform(data) {
        if (data != null && typeof data === "object" && "this" in data) {
          return {
            this: "transformed",
          };
        }
      },
    })],
  });

  if (!Array.isArray(result)) {
    expect.fail("result is not an array");
  }

  expect(result).toBeDefined();

  const firstResult = result[0];

  expect(firstResult).toBeDefined();
  expect(firstResult?.output).toBeDefined();
  expect(firstResult?.output[0]).toBeDefined();
  expect(firstResult?.output[0].code).toBeDefined();
  expect(firstResult?.output[0].code).toMatchSnapshot();
});
