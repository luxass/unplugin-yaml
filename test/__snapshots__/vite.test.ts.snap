// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handle multi document 1`] = `
"var data = [
  {
    "apiVersion": "batch/v1",
    "kind": "CronJob",
    "metadata": {
      "name": "first-cronjob",
      "namespace": "default"
    },
    "spec": {
      "schedule": "*/10 * * * *",
      "concurrencyPolicy": "Forbid",
      "successfulJobsHistoryLimit": 3,
      "failedJobsHistoryLimit": 1,
      "jobTemplate": {
        "spec": {
          "template": {
            "spec": {
              "containers": [
                {
                  "name": "first-job",
                  "image": "busybox:latest",
                  "command": [
                    "/bin/sh",
                    "-c",
                    'echo "First job running"; date'
                  ]
                }
              ],
              "restartPolicy": "OnFailure"
            }
          }
        }
      }
    }
  },
  {
    "apiVersion": "batch/v1",
    "kind": "CronJob",
    "metadata": {
      "name": "second-cronjob",
      "namespace": "default"
    },
    "spec": {
      "schedule": "0 0 * * *",
      "concurrencyPolicy": "Forbid",
      "successfulJobsHistoryLimit": 3,
      "failedJobsHistoryLimit": 1,
      "jobTemplate": {
        "spec": {
          "template": {
            "spec": {
              "containers": [
                {
                  "name": "second-job",
                  "image": "busybox:latest",
                  "command": [
                    "/bin/sh",
                    "-c",
                    'echo "Second job running"; date'
                  ]
                }
              ],
              "restartPolicy": "OnFailure"
            }
          }
        }
      }
    }
  }
];
console.log(data);
"
`;

exports[`handle transforms 1`] = `
"var data = {
  "this": "transformed"
};
console.log(data);
"
`;

exports[`handles yaml > expect yaml import to be a json object 1`] = `
"var data = {
  "pluginDir": "./plugins",
  "web": {
    "enabled": true
  },
  "logging": {
    "type": "stdout",
    "level": "info"
  }
};
console.log(data);
"
`;

exports[`handles yaml > expect yaml import to be a string 1`] = `
"const config = "pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n";
console.log(config);
"
`;

exports[`handles yml > expect yml import to be a json object 1`] = `
"var data = {
  "pluginDir": "./plugins",
  "web": {
    "enabled": true
  },
  "logging": {
    "type": "stdout",
    "level": "info"
  }
};
console.log(data);
"
`;

exports[`handles yml > expect yml import to be a string 1`] = `
"const config = "pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n";
console.log(config);
"
`;
